lib/Raven/Middleware/ContextInterfaceMiddleware.php

```diff
@@ -57,10 +85,15 @@ public function __invoke(Event $event, callable $next, ServerRequestInterface $r
         $runtimeContext = isset($payload['runtime_context']) ? $payload['runtime_context'] : [];
         $userContext = isset($payload['user_context']) ? $payload['user_context'] : [];
 
-        $event = $event->withTagsContext(array_merge($this->context->getTags(), $tagsContext), false)
-            ->withExtraContext(array_merge($this->context->getExtraData(), $extraContext), false)
-            ->withUserContext(array_merge($this->context->getUserData(), $userContext), false)
+        $event = $event->withTagsContext($this->tagsContext->toArray(), false)
+            ->withTagsContext($tagsContext, false)
+            ->withExtraContext($this->extraContext->toArray(), false)
```

Comment:
Why do we need to `toArray()` each time?

Reply:
Because in the `Event` class the context data is stored as a plain array. I could switch to an instance of the `Context class`, but it should probably not be the same instance as the one of the client or someone that adds an item to the context of an event would add it globally. WDYT?